### PR TITLE
Fix RS* alg and update dependencies

### PR DESCRIPTION
**Pull request by chulkilee**
#3 broke non HS* alg since the parameter structure is onlpy for HS* alg.

Also updated all dependencies.

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v5.10.0
```
Comment (mittsh) [inline .nvmrc:1]: Nice to have migrated it!

File: src/JsonWebTokenDynamicValue.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ class JsonWebTokenDynamicValue {
     const now = Math.floor((new Date()).getTime() / 1000);
     const header = {
       typ: "JWT",
-      alg: "HS256",
```
Comment (mittsh) [inline src/JsonWebTokenDynamicValue.js]: Ouch, good catch. I haven't picked this one the other day. Thx

File: src/JsonWebTokenDynamicValue.js [hunk 3]
```diff
@@ -58,9 +58,18 @@ class JsonWebTokenDynamicValue {
...
+
+    if(this.alg.substring(0, 2) == "HS") {
+      secret = this.signatureSecretIsBase64
+        ? {b64: jsrsasign.b64utob64(this.signatureSecret)}
+        : {utf8: this.signatureSecret}
+    } else {
+      secret = this.signatureSecretIsBase64
+        ? jsrsasign.b64utoutf8(this.signatureSecret)
+        : this.signatureSecret
+    }
```
Comment (mittsh) [inline src/JsonWebTokenDynamicValue.js:70]: Great you've fixed this, tbh I couldn't find the right format for HS and RS algs in the jsrsasign doc. Thanks for looking this up.

**Review by mittsh — COMMENTED** [review]
Really great fixes @chulkilee 
Many thanks for diving into this 👍 

I'll fork out of your PR to update a couple of things and will let you know.

**Comment by mittsh** [comment]
Oh and I was curious, why have you removed the `yarn.lock`?

**Comment by mittsh** [comment]
Merged @chulkilee 😄 
Many thanks!!

**Comment by chulkilee** [comment]
Does it use yarn? I didn't find any reference to it, so I just migrated it to npm v5 lock file (`package-lock.json`).

**Comment by mittsh** [comment]
Yeah I was using Yarn sometimes as it's much faster to install deps. In anyway, thank you for all the fixes :)

**Comment by chulkilee** [comment]
If this repo uses `yarn.lock` then it would be better to update `README.md` to use yarn instead of `npm install`

**Pull request merged**